File: anika.dovob/src/main/java/com/example/anika/homework/chapter7/SubscriptionItem.java
package com.example.anika.homework.chapter7;

import jakarta.persistence.*;

@Entity(name = "Ch7SubscriptionItem")
@Table(name = "subscription_item_ch7")
@DiscriminatorValue("SUBSCRIPTION")
public class SubscriptionItem extends Item {

    @Column(name = "billing_cycle_days", nullable = false)
    private int billingCycleDays;

    @Column(name = "trial_days", nullable = false)
    private int trialDays;

    protected SubscriptionItem() {
    }

    public SubscriptionItem(String name, long basePrice, int billingCycleDays, int trialDays) {
        super(name, basePrice);
        this.billingCycleDays = billingCycleDays;
        this.trialDays = trialDays;
    }

    public int getBillingCycleDays() {
        return billingCycleDays;
    }

    public int getTrialDays() {
        return trialDays;
    }
}

File: gracie.lee/src/main/kotlin/homework/chapter_7/SubscriptionItem.kt
package homework.chapter_7

import jakarta.persistence.*

@Entity(name = "Ch7SubscriptionItemKt")
@Table(name = "subscription_item_ch7_kt")
@DiscriminatorValue("SUBSCRIPTION") // DTYPE에 들어갈 값
class SubscriptionItem(
    @Column(nullable = false)
    var billingCycleDays: Int = 0,

    @Column(nullable = false)
    var trialDays: Int = 0
) : Item()
